### ISSUE TITLE
An argument named "transit_encryption_mode" is not expected here

### ISSUE THREAD

**Issue by MCharlot** [terraform-google-modules/terraform-google-memorystore#81]
### TL;DR

We are currently in Terraform 0.14.11 with a memorystore module in 3.1.0.
We want to update our version of Terraform to 1.1.4 and also update the memorystore module to 4.2.0.

When we perform a Terraform plan, we have this error.

### Expected behavior

Updated memorystore module to 4.2.0

### Observed behavior

│ Error: Unsupported argument
│
│   on .terraform/modules/memorystore-v5-standard/main.tf line 41, in resource "google_redis_instance" "default":
│   41:   transit_encryption_mode = var.transit_encryption_mode
│
│ An argument named "transit_encryption_mode" is not expected here.

### Terraform Configuration

```hcl
module "memorystore-v5-standard" {
  source                  = "terraform-google-modules/memorystore/google"
  version                 = "4.2.0"
  name                    = "memorystore-y-pp-1-v5"
  project                 = var.gcp_project_id
  labels                  = { foo = "bar", bar = "foo" }
  memory_size_gb          = var.memory_size_gb
  tier                    = var.tier
  region                  = var.region
  authorized_network      = data.google_compute_network.redis-network.id
  connect_mode            = var.connect_mode
  redis_version           = "REDIS_5_0"
  transit_encryption_mode = "DISABLED"
}
```

### Terraform Version

```sh
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/fluxcd/flux v0.3.1
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/hashicorp/aws v4.8.0
+ provider registry.terraform.io/hashicorp/google v3.72.0
+ provider registry.terraform.io/hashicorp/google-beta v3.72.0
+ provider registry.terraform.io/hashicorp/helm v2.5.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.9.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/template v2.1.2
+ provider registry.terraform.io/hashicorp/time v0.7.2
+ provider registry.terraform.io/hashicorp/tls v2.1.1
+ provider registry.terraform.io/hashicorp/vault v2.17.0
+ provider registry.terraform.io/integrations/github v4.23.0

Your version of Terraform is out of date! The latest version
is 1.1.7. You can update by downloading from https://www.terraform.io/downloads.html
```

**Comment by morgante** [comment]
You need to upgrade your Terraform provider version to a newer version.

**Comment by MCharlot** [comment]
I checked the dependencies ([doc](https://registry.terraform.io/modules/terraform-google-modules/memorystore/google/latest?tab=dependencies)) and it should be fine.


google (hashicorp/google) >= 3.53, < 5.0
google-beta (hashicorp/google-beta) >= 3.53, < 5.0

provider registry.terraform.io/hashicorp/google v3.72.0
provider registry.terraform.io/hashicorp/google-beta v3.72.0

**Comment by morgante** [comment]
You need at least v3.73. See the [provider docs](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#3730-june-21-2021).

**Comment by MCharlot** [comment]
Thanks @morgante for your response. I will do the necessary.

Perhaps the documentation should be updated to avoid misunderstandings.
